naming/dns_resolver_test.go

```diff
@@ -0,0 +1,304 @@
+package naming
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+)
+
+func newUpdate(op Operation, addr string) *Update {
```

Comment:
Use function can save field specifiers, e.g. Op:, Addr: each time we construct the Update. But I've changed to &Update{}, since I think the inline filed specifier make it more clear.